public/pages/Findings/containers/Findings/Findings.tsx

```diff
@@ -151,30 +218,74 @@ class Findings extends Component<FindingsProps, FindingsState> {
 
   onRefresh = async () => {
     await this.getNotificationChannels();
-    await this.getPlugins();
-    await this.getFindings();
+    if (this.state.selectedTabId === FindingTabId.DetectionRules) {
+      await this.getDetectionRulesFindings();
+    } else if (this.state.selectedTabId === FindingTabId.ThreatIntel) {
+      await this.getThreatIntelFindings();
+    }
     renderVisualization(this.generateVisualizationSpec(), 'findings-view');
   };
 
+  setStateForTab<T extends FindingTabId, F extends keyof FindingsState['findingStateByTabId'][T]>(
+    {
+      tabId,
+      field,
+      value,
+    }: { tabId: T; field: F; value: FindingsState['findingStateByTabId'][T][F] },
+    otherState?: Partial<Pick<FindingsState, keyof Omit<FindingsState, 'findingStateByTabId'>>>
+  ) {
+    this.setState({
+      ...(otherState as any),
+      findingStateByTabId: {
+        ...this.state.findingStateByTabId,
+        [tabId]: {
+          ...this.state.findingStateByTabId[tabId],
+          [field]: value,
+        },
+      },
+    });
+  }
+
   onStreamingFindings = async (findings: FindingItemType[]) => {
     const ruleIds = new Set<string>();
     findings.forEach((finding) => {
       finding.queries.forEach((rule) => ruleIds.add(rule.id));
     });
 
     await this.getRules(Array.from(ruleIds));
-    this.setState({ findings: [...this.state.findings, ...findings] });
-  }
+    this.setStateForTab({
+      tabId: FindingTabId.DetectionRules,
+      field: 'findings',
+      value: [...this.state.findingStateByTabId[FindingTabId.DetectionRules].findings, ...findings],
+    });
+  };
+
+  onStreamingThreatIntelFindings = async (findings: ThreatIntelFinding[]) => {
+    this.setStateForTab({
+      tabId: FindingTabId.ThreatIntel,
+      field: 'findings',
+      value: [...this.state.findingStateByTabId[FindingTabId.ThreatIntel].findings, ...findings],
+    });
+  };
 
   abortGetFindings = () => {
-    this.abortGetFindingsControllers.forEach(controller => {
+    this.abortGetFindingsControllers.forEach((controller) => {
       controller.abort();
     });
-  }
+  };
 
-  getFindings = async () => {
+  getDetectionRulesFindings = async () => {
     this.abortGetFindings();
-    this.setState({ loading: true, findings: [] });
+    this.setStateForTab(
```

Comment:
`updateTabState` ?

Reply:
since all it does is sets the state, I'm more inclined to keep this `setStateForTab`.